lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1039,55 +1079,153 @@ def configure_cpu_and_memory(self, vm_obj, vm_creation=False):
                 self.configspec.firmware = boot_firmware
                 self.change_detected = True
 
+    def get_configured_cdroms(self):
+        # cdroms {'ide': [{num: 0, cdrom: []}, {}], 'sata': [{num: 0, cdrom: []}, {}, ...]}
+        cdroms = {'ide': None, 'sata': None}
+        for cdrom_spec in self.params.get('cdrom'):
+            if 'type' not in cdrom_spec or cdrom_spec['type'] not in ['none', 'client', 'iso']:
+                self.module.fail_json(msg="cdrom.type is mandatory. Options are 'none', 'client', and 'iso'.")
+            if cdrom_spec['type'] == 'iso' and ('iso_path' not in cdrom_spec or not cdrom_spec['iso_path']):
+                self.module.fail_json(msg="cdrom.iso_path is mandatory when cdrom.type is set to iso.")
+            if 'controller_type' in cdrom_spec and cdrom_spec['controller_type'].lower() in ['sata', 'ide']:
+                cdrom_ctl = cdrom_spec['controller_type'].lower()
+            else:
+                self.module.fail_json(msg='CD-ROM controller type should be specified, valid value: ide or sata. %s' % cdrom_spec)
+            if 'controller_number' not in cdrom_spec or not isinstance(cdrom_spec['controller_number'], int):
+                self.module.fail_json(msg='CD-ROM attached controller bus number should be specified and integer.')
+            cdrom_ctl_num = cdrom_spec['controller_number']
+            if (cdrom_ctl == 'ide' and cdrom_ctl_num not in [0, 1]) or (cdrom_ctl == 'sata' and cdrom_ctl_num not in [0, 1, 2, 3]):
+                self.module.fail_json(msg='Invalid %s controller bus number: %d' % (cdrom_ctl, cdrom_ctl_num))
+            if cdroms[cdrom_ctl] is None:
+                cdroms[cdrom_ctl] = [{'num': cdrom_ctl_num, 'cdrom': [cdrom_spec]}]
+            else:
+                ctl_exist = False
```

Comment:
I'm not sure you need the `ctl_exist` variable.

Reply:
I need a variable to decide add the cdrom spec to the existing controller 'cdrom' list or add as a new controller.